### DISCUSSION TITLE
brew upgrade wipes Octave configuration files

### DISCUSSION THREAD

**Discussion by JordanSavoie** [Homebrew/discussions#3137] [Everyday usage]
Whenever Octave updates, Homebrew's script wipes the Octave startup script found at: 

/opt/homebrew/Cellar/octave/6.4.0_3/share/octave/site/m/startup/octaverc

Is there anything I can do to keep it from doing this?

**Comment by SMillerDev** [comment]
Shouldn't configuration files be in `/opt/homebrew/etc`?

**Reply by JordanSavoie** [reply]
It doesn't seem so. The Octave docs for the [startup files](https://octave.org/doc/v6.4.0/Startup-Files.html#Startup-Files) show the same location as it is on my installation. Additionally the [load path](https://octave.org/doc/v6.4.0/Manipulating-the-Load-Path.html) shows all locations within `Cellar/octave/[version]`.

**Reply by gromgit** [reply]
`octave` doesn't seem to provide a build-time option to configure site/version startup dirs, but it does have runtime environment overrides. From your first link:
> `octave-home/share/octave/site/m/startup/octaverc` [...] may be overridden by the environment variable `OCTAVE_SITE_INITFILE`.

That seems to be the only upstream-supported way to get what you want.

**Reply by JordanSavoie** [reply]
You're right. I'm a shell noob so I didn't process that until you pointed it out. Thank you.
